.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,48 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main, release/**, test-me-*]
+
+env:
+  PIP_REQUIREMENTS: requirements-dev.txt
+
+defaults:
+  run:
+    # the default default is:
+    #      bash --noprofile --norc -eo pipefail {0}
+    shell: bash --noprofile --norc -eo pipefail -ux {0}
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
```

Comment:
Could we pin to a sha rather than the version for these and use the latest versions?